lib/tasks/generateThemeDesignerResources.js

```diff
@@ -80,6 +153,39 @@ async function generateThemeDotTheming({workspace, combo, themeFolder}) {
 	return newDotThemingResource;
 }
 
+async function generateCssVariablesLess({workspace, combo, namespace}) {
+	let cssVariablesSourceLessResourcePattern;
+	if (namespace) {
+		// In case of a library only check for themes directly below the namespace
+		cssVariablesSourceLessResourcePattern = `/resources/${namespace}/themes/*/css_variables.source.less`;
+	} else {
+		// In case of a theme-library check for all "themes"
+		cssVariablesSourceLessResourcePattern = `/resources/**/themes/*/css_variables.source.less`;
+	}
+
+	const cssVariablesSourceLessResource = await workspace.byGlob(cssVariablesSourceLessResourcePattern);
+
+	const hasCssVariables = cssVariablesSourceLessResource.length > 0;
+
+	if (hasCssVariables) {
+		const cssVariablesLessResources = await Promise.all(
+			cssVariablesSourceLessResource.map((cssVariableSourceLess) => {
+				const themeFolder = posixPath.dirname(cssVariableSourceLess.getPath());
+				log.verbose(`Generating css_variables.less for theme ${themeFolder}`);
+				return createCssVariablesLessResource({
+					workspace, combo, themeFolder
+				});
+			})
+		);
+
+		return cssVariablesLessResources.map(async (resource) => {
```

Comment:
Isn't `Promise.all` missing here?

Reply:
Agreed on moving write logic to loop above

Reply:
Done